Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/servlet/SanitizationFilter.java

```diff
@@ -51,65 +48,74 @@
      */
     @Override
     public void destroy() {
         LOG.debug("Destroying my filter");
     }
 
-    /* (non-Javadoc)
-     * @see javax.servlet.Filter#doFilter(javax.servlet.ServletRequest, javax.servlet.ServletResponse, javax.servlet.FilterChain)
+    /*
+     * (non-Javadoc)
+     *
+     * @see javax.servlet.Filter#doFilter(javax.servlet.ServletRequest, javax.servlet.ServletResponse,
+     * javax.servlet.FilterChain)
      */
     @Override
-    public void doFilter(ServletRequest request, ServletResponse response, FilterChain filterChain) throws IOException,
-            ServletException {
-      LOG.debug("Prepare to interceptor request in Filter");
-      if (!(request instanceof HttpServletRequest)) {
-          filterChain.doFilter(request, response);
-          return;
-      }
-      if (request instanceof HttpServletRequest){
-          HttpServletRequest hrequest = (HttpServletRequest)request;
-          HttpServletResponse hresponse = (HttpServletResponse)response;
-          final String servletPath = hrequest.getServletPath();
-          LOG.debug("servlet Request: "+servletPath);
-          try {
-              //skip to check for js/image/css so that the custom error page can render properly
-              if (!servletPath.startsWith(ResourceHandler.RESOURCE_IDENTIFIER)){
-                  checkHeaders(hrequest);
-              }
-          } catch (SanitizationException e) {
-              LOG.error("Error in SanitilizeMessage {}: ",e.getLocalizedMessage(), e );
-              hrequest.getSession().invalidate();
-              hrequest.getServletContext().getRequestDispatcher(NavigationConstant.CUSTOM_ERROR_XHTML).forward(hrequest, hresponse);
-          } finally {
-              ESAPI.httpUtilities().clearCurrent();
-          }
-          if (hresponse.isCommitted()){
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain filterChain)
+            throws IOException, ServletException {
+        LOG.debug("Prepare to interceptor request in Filter");
+        if (!(request instanceof HttpServletRequest)) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+        if (request instanceof HttpServletRequest) {
+            HttpServletRequest hrequest = (HttpServletRequest) request;
+            HttpServletResponse hresponse = (HttpServletResponse) response;
+            final String servletPath = hrequest.getServletPath();
+            LOG.debug("servlet Request: " + servletPath);
+            try {
+                // skip to check for js/image/css so that the custom error page can render properly
+                if (!servletPath.startsWith(ResourceHandler.RESOURCE_IDENTIFIER)) {
+                    checkHeaders(hrequest);
+                }
+            } catch (SanitizationException e) {
+                LOG.error("Error in SanitilizeMessage {}: ", e.getLocalizedMessage(), e);
+                ESAPI.authenticator().logout();
```

Comment:
I think this will give you error since we have add any users into ESAPI api.  Please run locally to confirm.